gems/aws-crt/lib/aws-crt/http/message.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Aws
+  module Crt
+    # High level Ruby abstractions for CRT HTTP functionality
+    module Http
+      # HTTP Message (request)
+      class Message
+        include Aws::Crt::ManagedNative
+        native_destroy Aws::Crt::Native.method(
+          :http_message_release
+        )
+
+        def initialize(method, path, headers = {})
+          strings = [method, path] +
+                    headers.each_pair { |k, v| [k, v] }.flatten
+          blob = StringBlob.encode(strings)
+          blob_ptr = FFI::MemoryPointer.new(:char, blob.length)
+          blob_ptr.write_array_of_char(blob)
+
+          manage_native do
+            Aws::Crt::Native.http_message_new_from_blob(blob_ptr, blob.length)
+          end
+        end
+
+        def to_blob_strings
+          buf_out = Aws::Crt::Native::CrtBuf.new
+          Aws::Crt::Native.http_message_to_blob(native, buf_out)
+          StringBlob.decode(buf_out.to_blob)
+        end
+
+        def headers
+          blob_strings = to_blob_strings
+          blob_strings[2..blob_strings.length].each_slice(2).to_h
```

Comment:
`to_blob_strings[2..]` will work, you don't need the length. Also this could fail on `to_h` if `each_slice` returns one element (odd number of blob strings slice). Not sure if that matters.

Reply:
Doesn't work on older rubies :-(  length is required.

Reply:
But re: getting the wrong number of elements (ie, where each_slice(2) would return one element) - that should never happen if the blob string is for a valid message.  However, maybe makes sense to be defensive here and doing a check.

Reply:
What minimum version of Ruby are we supporting for CRT?

Reply:
2.5